### PR TITLE
Update SDK to ComputeCpp v0.5.1

### PR DESCRIPTION
**Pull request by DuncanMcBain**
More 1.2.1 changes have been integrated, though we're not 100% there
yet. Old code will continue to work but will show deprecation messages
(which I don't think are useful to see when building the SDK).

### CODE REVIEW

File: .travis/build_computecpp.sh [hunk 1]
```diff
@@ -5,7 +5,8 @@ set -ev
 ###########################
 # Get ComputeCpp
 ###########################
-wget --no-check-certificate https://computecpp.codeplay.com/downloads/computecpp-ce/0.5.0/ubuntu-14.04-64bit.tar.gz
+wget --no-check-certificate https://computecpp.codeplay.com/downloads/computecpp-ce/0.5.1/ubuntu-14.04-64bit.tar.gz
 tar -xzf ubuntu-14.04-64bit.tar.gz -C /tmp
+mv /tmp/ComputeCpp-CE-0.5.1-Ubuntu-14.04-64bit /tmp/computecpp
```
Comment (ProGTX) [inline .travis/build_computecpp.sh]: What if the folder already exists?

Comment (Ruyk) [inline .travis/build_computecpp.sh]: why don't you import the new Khronos SYCL Parallel STL travis script? It also test with Intel OpenCL when available

File: samples/ivka/ivka.cpp [hunk 2]
```diff
@@ -39,9 +39,11 @@ struct Bar : Foo {
 };
 
 using namespace cl::sycl;
+using m = cl::sycl::access::mode;
+using t = cl::sycl::access::target;
```
Comment (Ruyk) [inline samples/ivka/ivka.cpp]: please don't use one letter shortcuts, at least use mode and target

File: .travis/build_computecpp.sh (hunk 1 above)
Comment (DuncanMcBain) [reply .travis/build_computecpp.sh]: It won't, the machine is fresh every time.

Comment (DuncanMcBain) [reply .travis/build_computecpp.sh]: OK, I will add this in a separate CL I think. Thanks for the suggestion.

**Pull request merged**